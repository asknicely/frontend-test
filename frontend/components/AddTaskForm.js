import React from 'react';

function AddTaskForm(props) {
  const {
    handleAddTaskSubmit,
    handleChangeAddTaskField,
    addTaskField,
    addButtonIsActive,
  } = props;

  return (
    <form
      onSubmit={handleAddTaskSubmit}
      aria-label="Create a task"
      className="add-task-form"
    >
      <label htmlFor="description">Add a new todo</label>
      <input
        id="description"
        type="textbox"
        name="description"
        className="small-6 small-center"
        placeholder="e.g: Eat Donuts :)"
        value={addTaskField}
        onChange={handleChangeAddTaskField}
      />
      <button
        type="submit"
        className="btn btn-sm btn-primary"
        disabled={!addButtonIsActive}
      >
        Add
      </button>
    </form>
  );
}

export default AddTaskForm;
