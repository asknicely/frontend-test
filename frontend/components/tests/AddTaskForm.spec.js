import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import AddTaskForm from '../AddTaskForm';
import {
  handleChangeAddTaskField,
  handleAddTaskSubmit,
} from './helpers/helpers';

const addButtonIsActive = true;
const addTaskField = 'test';

describe('AddTaskForm', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<AddTaskForm />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render AddTaskForm correctly', () => {
    const wrapper = shallow(
      <AddTaskForm
        handleAddTaskSubmit={handleAddTaskSubmit}
        handleChangeAddTaskField={handleChangeAddTaskField}
        addTaskField={addTaskField}
        addButtonIsActive={addButtonIsActive}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
