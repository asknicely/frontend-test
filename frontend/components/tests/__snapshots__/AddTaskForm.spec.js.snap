// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTaskForm should render AddTaskForm correctly 1`] = `ShallowWrapper {}`;

exports[`AddTaskForm snapshot renders 1`] = `
<form
  aria-label="Create a task"
  className="add-task-form"
>
  <label
    htmlFor="description"
  >
    Add a new todo task
  </label>
  <input
    className="small-6 small-center"
    id="description"
    name="description"
    placeholder="e.g. Eat Donuts 🍩:)"
    type="textbox"
  />
  <button
    className="btn btn-sm btn-primary"
    disabled={true}
    type="submit"
  >
    Add
  </button>
</form>
`;
